lib/cn/cn_tree.c

```diff
@@ -651,11 +582,19 @@ cn_tree_insert_kvset(struct cn_tree *tree, struct kvset *kvset, uint level, uint
     struct cn_tree_node *    node;
     u64                      dgen;
 
-    assert(level <= 1);
-    assert(tree->ct_root);
-    assert(offset < tree->ct_root->tn_childc);
+    if (level == 0) {
+        node = tree->ct_root;
+    } else {
+        assert(level == 1);
+        list_for_each_entry(node,  &tree->ct_leaves, tn_link) {
+            if (node->tn_loc.node_offset == offset)
+                break;
+        }
+    }
 
-    node = level == 0 ? tree->ct_root : tree->ct_root->tn_childv[offset];
+    assert(node);
```

Comment:
It's a runtime sanity check on the goofy logic in the preceding few lines that finds the node with the given level and offset.  I imagine when node level and offset are removed this will change and the node will be passed as a function argument.